netbox/utilities/templatetags/helpers.py

```diff
@@ -137,6 +139,39 @@ def tzoffset(value):
     return datetime.datetime.now(value).strftime('%z')
 
 
+@register.filter(expects_localtime=True)
+def date_span(date_value):
+    """
+    Returns date as HTML span with short date format as the content and the
+    (long) date format as the title.
+    """
+    if not date_value:
+        return ''
+
+    if type(date_value) == str:
```

Comment:
I wouldn't think we'd ever want to apply this tag to a string; we should always be working with datetime objects, no?

Reply:
Here the date is a string: https://github.com/netbox-community/netbox/blob/develop/netbox/templates/inc/custom_fields_panel.html

Reply:
It'd be better to have the custom field return a date object than to accommodate string values in the template filter. Let's just omit custom field dates for now to keep this from getting any more complex.

Reply:
Fine for me, done.